app/src/main/java/org/openlake/sampoorna/presentation/features/periodtracker/TrackingFragment.kt

```diff
@@ -63,12 +67,25 @@ class TrackingFragment : Fragment(R.layout.fragment_tracking) {
             showDaysLeft(daysLeft)
 
             //saving the result
-            sharedPref.edit()?.putLong("expectedDate", futureDate.time)?.apply()
+            sharedPref.edit()
+                .putLong("expectedDate", futureDate.time)
+                .apply()
+
+            val intent = Intent(requireContext(), PeriodReceiver::class.java)
+            intent.putExtra("title", "Hey ${sharedPref.getString(Constants.Name,"")}!!")
```

Comment:
Use of force unwrapping (!!) is not considered a good practice and its use should be minimal. Instead, we can use the ?: operator to handle the null case.